monocle/db/queries.py

```diff
@@ -90,6 +85,8 @@ def generate_filter(repository_fullname, params):
         qfilter.append({"terms": {"author": authors}})
     if on_authors:
         qfilter.append({"terms": {"on_author": on_authors}})
+    if change_ids:
+        qfilter.append({"terms": {"change_id": change_ids}})
     if 'Change' in params['etype']:
```

Comment:
You should change this to:
if 'Change' in params['etype'] and len(params['etype"]) == 1:
to not enter in this condition for the new all_changes function

Reply:
I don't get it. I need this especially for all_changes.

Reply:
The exec flow does not enter in generate_events_filter due to that condition.

Reply:
done